integration/integration_test.go

```diff
@@ -508,12 +532,171 @@ func (s *BaseSuite) digHost(c *check.C, source, namespace, destination string) {
 	c.Assert(IP, checker.NotNil)
 }
 
-func contains(s []string, x string) bool {
-	for _, v := range s {
-		if x == v {
-			return true
+func (s *BaseSuite) getPod(c *check.C, name string) *corev1.Pod {
+	pod, err := s.client.GetKubernetesClient().CoreV1().Pods(testNamespace).Get(context.Background(), name, metav1.GetOptions{})
+	c.Assert(err, checker.IsNil)
+
+	return pod
+}
+
+func (s *BaseSuite) getService(c *check.C, name string) *corev1.Service {
+	svc, err := s.client.GetKubernetesClient().CoreV1().Services(testNamespace).Get(context.Background(), name, metav1.GetOptions{})
+	c.Assert(err, checker.IsNil)
+
+	return svc
+}
+
+func (s *BaseSuite) getTrafficTarget(c *check.C, name string) *access.TrafficTarget {
+	tt, err := s.client.GetAccessClient().AccessV1alpha1().TrafficTargets(testNamespace).Get(context.Background(), name, metav1.GetOptions{})
+	c.Assert(err, checker.IsNil)
+
+	return tt
+}
+
+func (s *BaseSuite) getTrafficSplit(c *check.C, name string) *split.TrafficSplit {
+	ts, err := s.client.GetSplitClient().SplitV1alpha2().TrafficSplits(testNamespace).Get(context.Background(), name, metav1.GetOptions{})
+	c.Assert(err, checker.IsNil)
+
+	return ts
+}
+
+func (s *BaseSuite) checkBlockAllMiddleware(c *check.C, config *dynamic.Configuration) {
+	c.Log("Checking if the block-all-middleware is blocks everything")
+
+	middleware := config.HTTP.Middlewares["block-all-middleware"]
+	c.Assert(middleware, checker.NotNil)
+
+	c.Assert(middleware.IPWhiteList.SourceRange, checker.HasLen, 1)
+	c.Assert(middleware.IPWhiteList.SourceRange[0], checker.Equals, "255.255.255.255")
+}
+
+func (s *BaseSuite) checkHTTPReadinessService(c *check.C, config *dynamic.Configuration) {
+	c.Log("Checking if the readiness service is correctly defined")
+
+	service := config.HTTP.Services["readiness"]
+	c.Assert(service, checker.NotNil)
+
+	c.Assert(service.LoadBalancer.Servers, checker.HasLen, 1)
+	c.Assert(service.LoadBalancer.Servers[0].URL, checker.Equals, "http://127.0.0.1:8080")
+}
+
+func (s *BaseSuite) checkHTTPServiceLoadBalancer(c *check.C, config *dynamic.Configuration, svc *corev1.Service, pods []*corev1.Pod) {
+	for _, port := range svc.Spec.Ports {
+		svcKey := fmt.Sprintf("%s-%s-%d", svc.Namespace, svc.Name, port.Port)
```

Comment:
Can't we use the provider keygen funcs to generate here?

Reply:
I thought about it too. But I would prefer not to expose functions from the `key.go` just for our integration tests. WDYT?

Reply:
Seems ok. Just didn't want to dup code.